blueocean-pipeline-scm-api/src/main/java/io/jenkins/blueocean/credential/SSHCredentialManager.java

```diff
@@ -0,0 +1,188 @@
+package io.jenkins.blueocean.credential;
+
+import com.cloudbees.jenkins.plugins.sshcredentials.impl.BasicSSHUserPrivateKey;
+import com.cloudbees.plugins.credentials.CredentialsProvider;
+import com.cloudbees.plugins.credentials.CredentialsScope;
+import com.cloudbees.plugins.credentials.CredentialsStore;
+import com.cloudbees.plugins.credentials.domains.Domain;
+import hudson.model.User;
+import hudson.remoting.Base64;
+import io.jenkins.blueocean.commons.ServiceException;
+import io.jenkins.blueocean.rest.impl.pipeline.credential.BlueOceanDomainRequirement;
+import io.jenkins.blueocean.rest.impl.pipeline.credential.BlueOceanDomainSpecification;
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.OutputStream;
+import java.math.BigInteger;
+import java.security.KeyFactory;
+import java.security.KeyPair;
+import java.security.KeyPairGenerator;
+import java.security.NoSuchAlgorithmException;
+import java.security.interfaces.RSAPrivateCrtKey;
+import java.security.interfaces.RSAPrivateKey;
+import java.security.interfaces.RSAPublicKey;
+import java.security.spec.InvalidKeySpecException;
+import java.security.spec.PKCS8EncodedKeySpec;
+import java.security.spec.RSAPublicKeySpec;
+import java.util.List;
+import javax.annotation.CheckForNull;
+import javax.annotation.Nonnull;
+import jenkins.model.Jenkins;
+
+/**
+ * Helper to deal with managing a Jenkins-created per-user SSH key
+ * @author kzantow
+ */
+public class SSHCredentialManager {
```

Comment:
`UserSSHKeyManager` (renamed) is used by a different plugin already, can't mark it as `@Restricted(NoExternalUse)`... adding a `reset` method - done.